### DISCUSSION TITLE
Cleanup - what happens when requesting a cleanup and why is it needed ?

### DISCUSSION THREAD

**Discussion by jesperordrup** [coollabsio/coolify#610] [Q&A] [answered]
Sometimes I find that the server has a large use of disk space. Today it said 46 GB used. So I clicked cleanup and now it says 15GB used. 

Thats nice, but:

I assume that it deletes cached stuff I dont need. But if I dont need it, why doesnt it happen automatically? 
Could it be that I sometimes dont want to clean up because it will make some stuff slower?
Is it deleting logs that I want to keep?

Suggestion:

Make it configurable to auto cleanup.
Make the feature transparent for what you delete and if it makes sense allow me to select parts I dont want to delete.

**Comment by Christopher-Hayes** [comment]
I'd suggest converting this issue into a discussion, since it's mainly an open question that others can learn from. And creating a new issue specifically for "make cleanup job configurable".

Edit: Oops, just realized this is already a discussion. The GitHub app makes them look identical. I'll leave this comment here to boost this discussion.

**Reply by dev-andremonteiro** [reply]
Replying to bump, could someone help us understand this ?

**Comment by peaklabs-dev** [comment, marked as answer]
@jesperordrup and @dev-andremonteiro It already cleans up automatically based on a cron job if you enable force docker cleanup in the server settings or as soon as you reach a certain disk space threshold for example 80% full. You can see what it does in the helper:

<img width="626" alt="image" src="https://github.com/user-attachments/assets/ca66fbd8-a066-4a19-90f4-7ad26f1e0ac1">
